config/oauth_visitors_test.go

```diff
@@ -0,0 +1,58 @@
+package config
+
+import (
+	"context"
+	"errors"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"golang.org/x/oauth2"
+)
+
+type mockTokenSource struct {
+	mockedTokenFunc func() (*oauth2.Token, error)
+}
+
+func (m mockTokenSource) Token() (*oauth2.Token, error) {
+	return m.mockedTokenFunc()
+}
```

Comment:
It should be outside the tests as it's being used by three tests, putting this inside will increase code duplication.

Reply:
it's fine to do it this way.